internal/services/signalr/signalr_service_custom_domain_resource.go

```diff
@@ -0,0 +1,231 @@
+				return fmt.Errorf("parsing custom certificate for %s: %+v", id, err)
+			}
+
+			existing, err := client.CustomDomainsGet(ctx, id)
+			if !response.WasNotFound(existing.HttpResponse) {
+				if err != nil {
+					return fmt.Errorf("retrieving %s: %v", id, err)
+				}
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			customDomainObj := signalr.CustomDomain{
+				Properties: signalr.CustomDomainProperties{
+					DomainName: customDomainSignalrServiceModel.DomainName,
+					CustomCertificate: signalr.ResourceReference{
+						Id: utils.String(customDomainSignalrServiceModel.SignalrCustomCertificateId),
+					},
+				},
+			}
+			if err := client.CustomDomainsCreateOrUpdateThenPoll(ctx, id, customDomainObj); err != nil {
+				return fmt.Errorf("creating signalR custom domain: %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (r CustomDomainSignalrServiceResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.SignalR.SignalRClient
+			id, err := signalr.ParseCustomDomainID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.CustomDomainsGet(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+				return fmt.Errorf("reading signalR custom domain %s: %+v", id, err)
+			}
+
+			if resp.Model == nil {
+				return fmt.Errorf("retrieving %s: got nil model", *id)
+			}
+
+			signalrServiceId := signalr.NewSignalRID(id.SubscriptionId, id.ResourceGroupName, id.SignalRName).ID()
+
+			state := CustomDomainSignalrServiceModel{
+				Name:             id.CustomDomainName,
+				DomainName:       resp.Model.Properties.DomainName,
+				SignalRServiceId: signalrServiceId,
+			}
+
+			if resp.Model.Properties.CustomCertificate.Id != nil {
+				signalrCustomCertId, err := signalr.ParseCustomCertificateIDInsensitively(*resp.Model.Properties.CustomCertificate.Id)
+				if err != nil {
+					return fmt.Errorf("parsing signalr custom cert id for %s: %+v", id, err)
+				}
+				state.SignalrCustomCertificateId = signalrCustomCertId.ID()
+			}
+
+			return metadata.Encode(&state)
+		},
+	}
+}
+
+func (r CustomDomainSignalrServiceResource) Delete() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.SignalR.SignalRClient
+
+			id, err := signalr.ParseCustomDomainID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			signalrId := signalr.NewSignalRID(id.SubscriptionId, id.ResourceGroupName, id.SignalRName)
+
+			if err := waitForSignalrServiceStatusToBeReady(ctx, metadata, signalrId); err != nil {
+				return fmt.Errorf("waiting for signalR service %s state to be ready error: %+v", signalrId, err)
+			}
```

Comment:
Thanks @stephybun for the comment, I see some different behaviors across the RPs, some of them were return model-nil error for the nil model, may I know if this "doesn't error return for the nil model" is the suggested practice for the future RP development?

Reply:
We need to make sure the parent resource signalR is not in any "in-progress" state such as "updating" by another operation before deleting it's sub-resource custom domain, otherwise, there would be error.

Reply:
Then it sounds like the parent resource is missing some waiting functionality in the create/update?

Reply:
@xiaxyi - have you addressed this comment yet?

Reply:
@xiaxyi - we should not be waiting for another resource to become ready in a delete. the parent resource should not return from its create/update/delete etc until it is truly finished doing its action ie the waiting should be happening elsewhere not here.